packages/core/src/dialects/abstract/query-generator.d.ts

```diff
@@ -158,6 +172,13 @@ export class AbstractQueryGenerator extends AbstractQueryGeneratorTypeScript {
     options?: ArithmeticQueryOptions,
   ): string;
 
+  createTableQuery(
+    tableName: TableNameOrModel,
+    // TODO: rename attributes to columns in the implementation when migrating to TS
```

Comment:
Much more importantly, when migrating to TypeScript, this should accept a map of attributes. Something like this:

```ts
createTableQuery(
  'users',
  { id: { type: DataTypes.STRING, allowNull: false } }
)
```

With a type that will probably be:

```ts
createTableQuery(model: Model): string
createTableQuery(
  tableName: TableName, 
  columns: { [columnName: string]: Literal | ColumnOptions },
): string
```

Reply:
Updated the TODO and linked your comment